src/Components/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL/AspireEFPostgreSqlExtensions.cs

```diff
@@ -117,12 +121,52 @@ void ConfigureDbContext(DbContextOptionsBuilder dbContextOptionsBuilder)
 
         void ConfigureRetry()
         {
-            if (!settings.Retry)
+#pragma warning disable EF1001 // Internal EF Core API usage.
+            if (settings.Retry || settings.CommandTimeout.HasValue)
             {
-                return;
+                builder.PatchServiceDescriptor<TContext>(optionsBuilder => optionsBuilder.UseNpgsql(options =>
+                {
+                    var extension = optionsBuilder.Options.FindExtension<NpgsqlOptionsExtension>();
+
+                    if (settings.Retry)
+                    {
+                        var executionStrategy = extension?.ExecutionStrategyFactory?.Invoke(new ExecutionStrategyDependencies(null!, optionsBuilder.Options, null!));
```

Comment:
@sebastienros just to confirm, you're referring to this right?

/cc @AndriySvyryd

Reply:
Yes, `new ExecutionStrategyDependencies(null!` I don't know any other way to get the actual `ExtensionStrategy`, and our tests are actually passing a concrete DbContext because we own the type and know how to construct it. There this code makes me sad, the ctor in other EF providers could definitely guard against it, right?

Reply:
Yes, there's no other way for 8. But, I assume all of this will be removed for the next version, right? You'd just need to make sure that the user calls Enrich before AddDbContext, so all Aspire configuration can be silently overridden.

For SqlServer you can get this behavior in 8 by calling `UseAzureSqlDefaults` instead of `EnableRetryOnFailure`

Reply:
> You'd just need to make sure that the user calls Enrich before AddDbContext

I don't think that's the API experience we are looking for. IMO, the current experience makes more sense:

1. AddDbContext
2. EnrichDbContext

If you flip it the other way around, it doesn't really make sense. How can you "Enrich" something before you add it?

Reply:
Ok, perhaps the question is whether Enrich just provides a recommended default configuration or whether it should be authorative over anything specified in `AddDbContext`.

Personally, I don't see why it should override or throw for any conflicting configuration. A suppressible warning seems more appropriate.